src/packagedcode/spec.py

```diff
@@ -0,0 +1,158 @@
+            lines = data.readlines()
+
+        spec_data = {}
+
+        for line in lines:
+            if '.name' in line:
+                name = line.rpartition('=')
+                spec_data['name'] = get_stripped_data(name[2])
+            elif '.version' in line and '.version.' not in line:
+                version = line.rpartition('=')
+                spec_data['version'] = get_stripped_data(version[2])
+            elif '.license' in line:
+                license_type = line.rpartition('=')
+                spec_data['license'] = get_stripped_data(license_type[2])
+            elif '.source' in line and '.source_files' not in line:
+                source = re.sub(r'/*.*source.*?>', '', line)
+                stripped_source = re.sub(r',.*', '', source)
+                spec_data['source'] = get_stripped_data(stripped_source)
+            elif '.email' in line:
+                emails = line.rpartition('=')
+                stripped_emails = get_stripped_data(emails[2])
+                stripped_emails = stripped_emails.strip()
+                stripped_emails = stripped_emails.split(',')
+                spec_data['email'] = stripped_emails
+            elif '.author' in line:
+                authors = re.sub(r'/*.*author.*?=', '', line)
+                stripped_authors = get_stripped_data(authors)
+                stripped_authors = stripped_authors.replace(' => ', "=>")
+                stripped_authors = stripped_authors.strip()
+                stripped_authors = stripped_authors.split(',')
+                spec_data['author'] = stripped_authors
+            elif '.summary' in line:
+                summary = line.rpartition('=')
+                spec_data['summary'] = get_stripped_data(summary[2])
+            elif '.description' in line:
+                if location.endswith('.gemspec'):
+                    # FIXME: description can be in single or multi-lines
+                    # There are many different ways to write description.
+                    desc = line.rpartition('=')
+                    spec_data['description'] = get_stripped_data(desc[2])
+                else:
+                    spec_data['description'] = get_description(line)
+            elif '.homepage' in line:
+                homepage_url = line.rpartition('=')
+                spec_data['homepage_url'] = get_stripped_data(homepage_url[2])
+
+        parser = GemfileParser(location)
+        deps = parser.parse()
+        dependencies = OrderedDict()
+        for key in deps:
+            if deps[key]:
+                for dependency in deps[key]:
+                    dependencies[dependency.name] = (dependency.requirement).split(',')
+        spec_data['dependencies'] = dependencies
+
+        return spec_data
+
+
+def get_stripped_data(line):
+    """
+    Return line after removing unnecessary special character and space.
+    """
+    if '#' in line:
+        line = line[:line.index('#')]
+    stripped_data = line.strip()
```

Comment:
what about chaining the calls instead?
may be more or less:
```Python
stripped_data = (
    line.strip()
    .replace("'", '')
    .replace('"', '')
    .replace('{', '')
    .replace('}', '')
    .replace('[', '')
    .replace(']', '')
    .replace('%q', '')
    .strip()
)
```
or a loop of sorts?
```Python
stripped_data = line.strip()
for strippable in ("'",'"', '{', '}', '[', ']', '%q',):
    stripped_data = stripped_data.replace(strippable, '')

return stripped_data.strip()
```